leader-election-impl/src/main/java/com/palantir/leader/PaxosLeaderElectionService.java

```diff
@@ -126,8 +136,8 @@ private void proposeLeadershipOrWaitForBackoff(LeadershipState currentState) thr
             return;
         }
 
-        if (pingLeader(currentState.greatestLearnedValue()).isSuccessful()) {
-            Thread.sleep(updatePollingRateInMs);
+        if (isSuccessfulPing(pingLeader(currentState.greatestLearnedValue()))) {
```

Comment:
I think this name is misleading, "isSuccessfulPing suggests that you're just checking some invariant, but in fact you're also caching the result of the lookup. I would say that doing the caching in pingLeader is more appropriate.

Reply:
fair, bit difficult to describe since I want to say "is this successful and if it is cache it, but I don't want to know whether or not the cache operation is successful"

Reply:
I feel like pingLeader is basically where this should happen, then you don't need a name.